google/cloud/bigquery/job.py

```diff
@@ -588,7 +587,7 @@ def _check_resource_config(cls, resource):
 
     def to_api_repr(self):
         """Generate a resource for the job."""
-        raise NotImplementedError("Abstract")
+        return copy.deepcopy(self._properties)
 
     _build_resource = to_api_repr  # backward-compatibility alias
```

Comment:
This all LGTM, but I'm curious what this backwards compatibility is for here. It's not clear to me why we were specifically testing to make sure this would raise NotImplementedError before.

Reply:
> It's not clear to me why we were specifically testing to make sure this would raise NotImplementedError before.

I think that was solely to make sure the coverage tests pass. We're still doing a bit of funny business in the `*Job` classes, in that most (all?) subclasses omit the `statistics` property from their `to_api_repr`.

> I'm curious what this backwards compatibility is for here.

I don't remember, actually. I see a few hits for `_build_resource` in search. https://github.com/googleapis/python-bigquery/search?p=1&q=_build_resource I think `_build_resource` is usually the name we use for "update" methods that need to take a field mask and only populate certain fields. Jobs don't support update, so not sure why we ever had one.